engines/gob/inter_v7.cpp

```diff
@@ -263,6 +352,218 @@ void Inter_v7::o7_loadFunctions() {
 	_vm->_game->loadFunctions(tot, flags);
 }
 
+void Inter_v7::copyFile(const Common::String &sourceFile, const Common::String &destFile) {
+	SaveLoad::SaveMode mode1 = _vm->_saveLoad->getSaveMode(sourceFile.c_str());
+	SaveLoad::SaveMode mode2 = _vm->_saveLoad->getSaveMode(destFile.c_str());
+
+	if (mode2 == SaveLoad::kSaveModeIgnore || mode2 == SaveLoad::kSaveModeExists)
+		return;
+	else if (mode2 == SaveLoad::kSaveModeSave) {
+		if (mode1 == SaveLoad::kSaveModeNone) {
+			Common::SeekableReadStream *stream = _vm->_dataIO->getFile(sourceFile);
+			if (!stream)
+				return;
+
+			int32 size = stream->size();
+			byte *data = new byte[size];
+			stream->read(data, size);
+
+			_vm->_saveLoad->saveFromRaw(destFile.c_str(), data, size, 0);
+			delete[] data;
+			delete stream;
+		} else if (mode1 == SaveLoad::kSaveModeSave) {
+			_vm->_saveLoad->copySaveGame(sourceFile.c_str(), destFile.c_str());
+		} else
+			warning("o7_copyFile(): unsupported mode %d for source \"%s\" while copying to \"%s\" ",
+					mode1,
+					sourceFile.c_str(),
+					destFile.c_str());
+	}
+	else
+		warning("Attempted to write to file \"%s\" while copying from \"%s\"",
+				destFile.c_str(),
+				sourceFile.c_str());
+
+}
+
+void Inter_v7::o7_copyFile() {
+	Common::String path1 = _vm->_game->_script->evalString();
+	Common::String path2 = _vm->_game->_script->evalString();
+
+	debugC(2, kDebugFileIO, "Copy file \"%s\" to \"%s", path1.c_str(), path2.c_str());
+
+	Common::String file1 = getFile(path1.c_str(), false);
+	Common::String file2 = getFile(path2.c_str(), false);
+	if (file1.equalsIgnoreCase(file2)) {
+		warning("o7_copyFile(): \"%s\" == \"%s\"", path1.c_str(), path2.c_str());
+		return;
+	}
+
+	copyFile(file1, file2);
+}
+
+void Inter_v7::o7_deleteFile() {
+	Common::String file =_vm->_game->_script->evalString();
+
+	debugC(2, kDebugFileIO, "Delete file \"%s\"", file.c_str());
+
+	bool isPattern = file.contains('*') || file.contains('?');
+	Common::List<Common::String> files;
+	if (isPattern) {
+		files = _vm->_saveLoad->getFilesMatchingPattern(file.c_str());
+		debugC(2, kDebugFileIO, "Delete file matching pattern \"%s\" (%d matching file(s))", file.c_str(), files.size());
+		for (const Common::String &matchingFile : files)
+			debugC(5, kDebugFileIO, "Matching file: %s", matchingFile.c_str());
+	}
+	else {
+		files.push_back(file);
+		debugC(2, kDebugFileIO, "Delete file \"%s\"", file.c_str());
+	}
+
+	if (_vm->getGameType() == kGameTypeAdibou2
+		&& isPattern
+		&& file.hasPrefix("DATA\\??????")) {
+		// WORKAROUND a bug in original game: files APPLI_<N>.INF and CRITE_<N>.INF should not be deleted when removing character <N>
```

Comment:
maybe also the "EnableEnhancement" flag mentioned above?

Reply:
I believe this flag is no used yet in GOB, indeed. I will add it to the "free bananas" funny fix above, but I am a bit more reluctant for this one, as it leads to some unwanted data destruction, which cannot be undone (it erases the completion statistics in applications, if I remember well).

Reply:
I have finally left the "enableEnhancement" flag for future work (I thought it was already available as a global feature, but it seems to be only in SCUMM now).